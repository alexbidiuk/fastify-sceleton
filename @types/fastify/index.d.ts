import { Logger } from 'fastify';

declare module 'fastify' {
  export interface FastifyInstance {
    swagger: any;
  }

  export interface FastifyReply<HttpResponse> {
    SSEClientId: string;
    log: Logger;
  }

  export interface FastifyRequest<HttpRequest> {
    user: { name: string; id: string };
  }
}
