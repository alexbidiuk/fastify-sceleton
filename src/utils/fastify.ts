import { FastifyInstance } from 'fastify';

const registerRoutes = (
  fastifyInstance: FastifyInstance,
  routesMap: object,
) => {
  Object.values(routesMap).forEach(({ method, url, options, controller }) =>
    fastifyInstance[method](url, options, controller),
  );
};

export { registerRoutes };
