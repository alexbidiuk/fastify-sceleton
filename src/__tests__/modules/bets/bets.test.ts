import * as fastify from 'fastify';

import betsRoutesPlugin from '../../../modules/bets/bets.routes.plugin';

describe('/bets', () => {
  let server: fastify.FastifyInstance;

  beforeEach(async () => {
    server = fastify({});
    server.register(betsRoutesPlugin);
    await server.ready();

    jest.clearAllMocks();
  });

  it('POST returns 404', async (done) => {
    const response = await server.inject({ method: 'POST', url: '/bets/1' });
    expect(response.statusCode).toEqual(404);
    expect(response.payload).toMatchSnapshot();

    done();
  });
});
