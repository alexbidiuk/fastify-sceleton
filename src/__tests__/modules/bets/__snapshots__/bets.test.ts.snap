// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/bets POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/bets/1 not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
